easybuild/easyconfigs/k/Kent_tools/Kent_tools-461-GCC-12.2.0.eb

```diff
@@ -0,0 +1,44 @@
+easyblock = 'MakeCp'
+
+name = 'Kent_tools'
+version = '461'
+
+homepage = 'https://genome.cse.ucsc.edu/'
+description = """Kent utilities: collection of tools used by the UCSC genome browser."""
+
+toolchain = {'name': 'GCC', 'version': '12.2.0'}
+source_urls = [
+    'https://hgdownload.cse.ucsc.edu/admin/exe/',
+    'https://hgdownload.cse.ucsc.edu/admin/exe/userApps.archive/',
+]
+sources = ['userApps.v%(version)s.src.tgz']
+checksums = [
```

Comment:
@VictorGoitea is there a reason why you've removed the previous OpenSSL patch?

Reply:
also would like to know reason to remove openssl patch.  Speculating, maybe the patch for v442 is out of date?

Reply:
I did not observe the SSL picked up by the OS and not by EasyBuild (patch's aim). Moreover when trying to use the patch I get error:

```== 2024-06-25 11:03:25,878 easyblock.py:3958 INFO Running method extract_step part of step source                                           [80/345]
== 2024-06-25 11:03:25,878 easyblock.py:2576 INFO Unpacking source userApps.v461.src.tgz                                                            
== 2024-06-25 11:03:25,879 run.py:247 INFO running cmd: tar xzf /home/ucloud/.local/easybuild/sources/k/Kent_tools/userApps.v461.src.tgz            
== 2024-06-25 11:03:27,351 build_log.py:267 INFO ... (took 1 secs)                                                                                  
== 2024-06-25 11:03:27,352 build_log.py:267 INFO patching...                                                                                        
== 2024-06-25 11:03:27,353 easyblock.py:3950 INFO Starting patch step                                                                               
== 2024-06-25 11:03:27,353 easyconfig.py:1702 INFO Generating template values...                                                                    
== 2024-06-25 11:03:27,354 easyconfig.py:1721 INFO Template values: arch='x86_64', bitbucket_account='kent_tools', builddir='/home/ucloud/.local/eas
ybuild/build/Kent_tools/461/GCC-12.2.0', github_account='kent_tools', installdir='/work/easybuild/software/Kent_tools/461-GCC-12.2.0', module_name='
Kent_tools/461-GCC-12.2.0', name='Kent_tools', nameletter='K', nameletterlower='k', namelower='kent_tools', parallel='60', start_dir='None', sysroot
='', toolchain_name='GCC', toolchain_version='12.2.0', version='461', version_major='461', versionprefix='', versionsuffix=''                       
== 2024-06-25 11:03:27,354 easyblock.py:3958 INFO Running method patch_step part of step patch                                                      
== 2024-06-25 11:03:27,354 easyblock.py:2594 INFO Applying patch Kent_tools-442_openssl.patch                                                       
== 2024-06-25 11:03:27,356 run.py:247 INFO running cmd: patch -p1 -i /home/ucloud/.local/easybuild/easyconfigs/k/Kent_tools/Kent_tools-442_openssl.p
atch 
== 2024-06-25 11:03:27,366 run.py:700 WARNING Found 3 potential errors (some may be harmless) in output of patch -p1 -i /home/ucloud/.local/easybuil
d/easyconfigs/k/Kent_tools/Kent_tools-442_openssl.patch:
        Hunk #1 FAILED at 133.
        Hunk #2 FAILED at 142.
        2 out of 2 hunks FAILED -- saving rejects to file kent/src/inc/common.mk.rej
== 2024-06-25 11:03:27,409 build_log.py:171 ERROR EasyBuild crashed with an error (at easybuild/base/exceptions.py:126 in __init__): Couldn't apply 
patch file /home/ucloud/.local/easybuild/easyconfigs/k/Kent_tools/Kent_tools-442_openssl.patch. Process exited with code 1: patching file kent/src/i
nc/common.mk
Hunk #1 FAILED at 133.
Hunk #2 FAILED at 142.
2 out of 2 hunks FAILED -- saving rejects to file kent/src/inc/common.mk.rej
 (at easybuild/tools/filetools.py:1630 in apply_patch)
== 2024-06-25 11:03:27,410 filetools.py:2013 INFO Removing lock /work/easybuild/software/.locks/_work_easybuild_software_Kent_tools_461-GCC-12.2.0.l
ock...
== 2024-06-25 11:03:27,413 filetools.py:383 INFO Path /work/easybuild/software/.locks/_work_easybuild_software_Kent_tools_461-GCC-12.2.0.lock succes
sfully removed.
== 2024-06-25 11:03:27,413 filetools.py:2017 INFO Lock removed: /work/easybuild/software/.locks/_work_easybuild_software_Kent_tools_461-GCC-12.2.0.l
ock
== 2024-06-25 11:03:27,413 easyblock.py:4285 WARNING build failed (first 300 chars): Couldn't apply patch file /home/ucloud/.local/easybuild/easycon
figs/k/Kent_tools/Kent_tools-442_openssl.patch. Process exited with code 1: patching file kent/src/inc/common.mk
Hunk #1 FAILED at 133.
Hunk #2 FAILED at 142.
2 out of 2 hunks FAILED -- saving rejects to file kent/src/inc/common.mk.rej

== 2024-06-25 11:03:27,414 easyblock.py:328 INFO Closing log for application name Kent_tools version 461
```

This is because the file `common.mk` is different and the patch does not find those lines anymore. In the new `common.mk`, I did not find lines of code referring to the same as in the patch but maybe I overlook the matter

Can you @verdurin / @emdrago take a look to the new `common.mk` and give your opinion? I'm also curious if a patch would be necessary.

Reply:
@VictorGoitea sorry for not getting back to this sooner.
Could you post the relevant sections of `common.mk` here so we have a record in the PR?

Reply:
Actually I'm working on a PR for `468` which doesn't have the patch, and checking after the build has finished, it appears the linking is in order i.e. all using the `OpenSSL` module.

Reply:
Yes, my build of this PR shows binaries are linked to the EB-provided `OpenSSL`, so the patch is not needed.